### ISSUE TITLE
test_cftp shouldn't manually iterate the reactor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10786]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1180|
|Type|defect|
|Created|2005-09-19 16:27:45Z|

<details><summary>Searchable metadata</summary>

```
trac-id__1180 1180
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__tests tests
time__1127147265000000 1127147265000000
changetime__1159219625000000 1159219625000000
version__ 
owner__ 
cc__exarkun cc__spiv cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
By calling reactor.iterate() in a loop in combination with timeouts on output,
test_cftp is made much more fragile than needs be.

In particular, _getBatchOutput should return a Deferred instead of attempting to
return a synchronous value.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml set owner to @jml|
|-|-|

This will be deprecated behaviour soon, and will raise errors soon after that.

[branches/unclean-errors-2091-2](branches/unclean-errors-2091-2) provides an alternative implementation of `_getCmdResult` (one of the offenders).

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_3"></a>|@dreid removed owner|
|-|-|

[branches/getBatchOutput-1180](branches/getBatchOutput-1180) addresses problems with both of the getBatchOutput implementations.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml set owner to @dreid|
|-|-|

Looks good. Please add docstrings to testBatchFile, testError and testIgnoredError, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_5"></a>|@dreid commented|
|-|-|

I was afraid you were going to ask that, z3p do you mind explaining what these tests are actually doing?

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_6"></a>|@jml commented|
|-|-|

z3p has added docstrings, woot!

dreid, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_7"></a>|@dreid set status to `closed`|
|-|-|

(In [18283]) Remove usages of Deferred.setTimeout and reactor.iterate

Authors: dreid, z3p
Reviewer: jml
Fixes [#10786](https://github.com/twisted/twisted/issues/10786)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
